# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/visibility.rs
# Commit: e24ed89246a4
# Full Hash: e24ed89246a4a6c6b4a567ec412f0e3295b4570d
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-04-26 09:46:09
# Regressor Bug: 1766017
# File Overlap Count: 1
# Description:
#   Bug 1766017 - Do a better job of clipping off-screen surface allocation sizes r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D144534
# ==============================================================================

diff -r cdc42b7822f1 -r e24ed89246a4 gfx/wr/webrender/src/visibility.rs
--- a/gfx/wr/webrender/src/visibility.rs	Mon Apr 25 22:28:35 2022 +0000
+++ b/gfx/wr/webrender/src/visibility.rs	Mon Apr 25 23:40:40 2022 +0000
@@ -177,7 +177,9 @@
                 frame_context.spatial_tree,
             );
 
-            let surface_local_rect = surfaces[raster_config.surface_index.0].local_rect.cast_unit();
+            let surface_local_rect = surfaces[raster_config.surface_index.0]
+                .unclipped_local_rect
+                .cast_unit();
 
             // Let the picture cache know that we are pushing an off-screen
             // surface, so it can treat dependencies of surface atomically.